### PR TITLE
mgr/nfs: minor organizational cleanups

### PR DESCRIPTION
**Pull request by phlogistonjohn**
The changes in this PR try to clean up the mgr nfs module code and organization a bit. This was going to be part of a larger set of changes to increase code coverage by the tests but that set of changes would have been huge.
Instead, this series makes some more modest changes aiming at making the module more clearer, especially, it is hoped, to newcomers. This series is also the precursor to one more set of patches that will extend the tests to cover nfs-export management without an orchestration module.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/pybind/mgr/nfs/ganesha_conf.py [hunk 1]
```diff
@@ -10,6 +10,42 @@
...
+
+def _format_val(block_name: str, key: str, val: str) -> str:
+    if isinstance(val, list):
+        return ', '.join([_format_val(block_name, key, v) for v in val])
+    if isinstance(val, bool):
+        return str(val).lower()
+    if isinstance(val, int) or (block_name == 'CLIENT'
+                                and key == 'clients'):
+        return '{}'.format(val)
+    return '"{}"'.format(val)
```
Comment (mgfritch) [inline src/pybind/mgr/nfs/ganesha_conf.py:25]: (nit) the diff is somewhat hard to read when these are defined at the top and the `format_block` is at the bottom, should we move these down so all the related funcs are located together?

**Review by mgfritch — APPROVED** [review]
generally the refactor is correct, just a few suggestions ..

**Review by adk3798 — APPROVED** [review]

File: src/pybind/mgr/nfs/ganesha_conf.py (hunk 1 above)
Comment (phlogistonjohn) [reply src/pybind/mgr/nfs/ganesha_conf.py:25]: Thanks. I will keep that in mind for next time.

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2022-05-04_19:11:28-orch:cephadm-wip-adk-testing-2022-05-04-1135-distro-basic-smithi/

green

**Pull request merged**